### PR TITLE
refactor(cbr): refactor resources and related tests

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:

Refactor vault and policy resources and related tests.
- repair prepaid issue (vault).
- adjust resource's functions.
- adjust acc tests.
- all parameters support description.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

##
In some tests, because the binding (unbinding) of associated resources takes a long time and there is no checking logic after the binding (unbinding) action, the associated resource queried by the resource after the resource update is completed is inconsistent with the expected associated resource

step1:  associate two volumes to manage backups and check configuration.
step2:  dissociate a volume and associate another and check configuration.

expect:
    - step1: volume number: 2
    - step2: volume number: 2

actual:
    - step1: volume number: 2
    - step2: volume number: 3

A disk is in the state of unbinding, which is considered as a bound volume.

##
When creating the prepaid vault, the resource ID is mistakenly passed into the public function of the order check as the order ID, resulting in an exception.

##
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. all parameters support description.
2. refactor vault and policy resources.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAcc'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAcc -timeout 360m -parallel 4
=== RUN   TestAccDataBackup_basic
=== PAUSE TestAccDataBackup_basic
=== RUN   TestAccDataVaults_backupServer
=== PAUSE TestAccDataVaults_backupServer
=== RUN   TestAccDataVaults_replicationServer
=== PAUSE TestAccDataVaults_replicationServer
=== RUN   TestAccDataVaults_volume
=== PAUSE TestAccDataVaults_volume
=== RUN   TestAccDataVaults_backupTurbo
=== PAUSE TestAccDataVaults_backupTurbo
=== RUN   TestAccDataVaults_replicationTurbo
=== PAUSE TestAccDataVaults_replicationTurbo
=== RUN   TestAccPolicy_basic
=== PAUSE TestAccPolicy_basic
=== RUN   TestAccPolicy_retention
=== PAUSE TestAccPolicy_retention
=== RUN   TestAccPolicy_replication
=== PAUSE TestAccPolicy_replication
=== RUN   TestAccVault_backupServer
=== PAUSE TestAccVault_backupServer
=== RUN   TestAccVault_replicationServer
=== PAUSE TestAccVault_replicationServer
=== RUN   TestAccVault_prePaidServer
=== PAUSE TestAccVault_prePaidServer
=== RUN   TestAccVault_volume
=== PAUSE TestAccVault_volume
=== RUN   TestAccVault_backupTurbo
=== PAUSE TestAccVault_backupTurbo
=== RUN   TestAccVault_replicationTurbo
=== PAUSE TestAccVault_replicationTurbo
=== RUN   TestAccVault_AutoBind
=== PAUSE TestAccVault_AutoBind
=== RUN   TestAccVault_bindPolicies
=== PAUSE TestAccVault_bindPolicies
=== CONT  TestAccDataBackup_basic
=== CONT  TestAccVault_backupServer
=== CONT  TestAccDataVaults_replicationTurbo
=== CONT  TestAccPolicy_retention
--- PASS: TestAccPolicy_retention (16.57s)
=== CONT  TestAccDataVaults_backupTurbo
--- PASS: TestAccDataVaults_replicationTurbo (17.19s)
=== CONT  TestAccDataVaults_volume
--- PASS: TestAccDataVaults_volume (331.47s)
=== CONT  TestAccDataVaults_replicationServer
--- PASS: TestAccDataVaults_replicationServer (13.61s)
=== CONT  TestAccDataVaults_backupServer
--- PASS: TestAccVault_backupServer (373.55s)
=== CONT  TestAccPolicy_basic
--- PASS: TestAccPolicy_basic (13.93s)
=== CONT  TestAccVault_backupTurbo
--- PASS: TestAccDataVaults_backupTurbo (518.06s)
=== CONT  TestAccVault_bindPolicies
--- PASS: TestAccVault_bindPolicies (32.46s)
=== CONT  TestAccVault_AutoBind
--- PASS: TestAccVault_AutoBind (18.12s)
=== CONT  TestAccVault_replicationTurbo
--- PASS: TestAccVault_replicationTurbo (13.90s)
=== CONT  TestAccVault_prePaidServer
--- PASS: TestAccDataBackup_basic (680.63s)
=== CONT  TestAccVault_volume
--- PASS: TestAccDataVaults_backupServer (340.27s)
=== CONT  TestAccVault_replicationServer
--- PASS: TestAccVault_replicationServer (14.41s)
=== CONT  TestAccPolicy_replication
--- PASS: TestAccPolicy_replication (7.19s)
--- PASS: TestAccVault_volume (365.30s)
--- PASS: TestAccVault_prePaidServer (522.45s)
--- PASS: TestAccVault_backupTurbo (999.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       1386.881s
```

### CODE REVIEW

File: docs/data-sources/cbr_vaults.md [hunk 1]
```diff
@@ -20,66 +20,67 @@ data "huaweicloud_cbr_vaults" "test" {
...
+* `protection_type` - (Optional, String) Specifies the protection type of the vault.
   The valid values are **backup** and **replication**. Vaults of type **disk** don't support **replication**.
 
 * `size` - (Optional, Int) Specifies the vault sapacity, in GB. The valid value range is `1` to `10,485,760`.
 
 * `auto_expand_enabled` - (Optional, Bool) Specifies whether to enable automatic expansion of the backup protection
   type vault. Default to **false**.
 
-* `enterprise_project_id` - (Optional, String) Specifies a unique ID in UUID format of enterprise project.
+* `enterprise_project_id` - (Optional, String) Specifies the ID of the enterprise project to which the vault belongs.
```
Comment (houpeng80) [inline docs/data-sources/cbr_vaults.md:49]: add ForceNew

File: docs/resources/cbr_policy.md [hunk 2]
```diff
@@ -52,20 +52,21 @@ resource "huaweicloud_cbr_policy" "test" {
...
+* `type` - (Required, String, ForceNew) Specifies the protection type of the policy.
   Valid values are **backup** and **replication**.
   Changing this will create a new policy.
 
-* `backup_cycle` - (Required, List) Specifies the scheduling rule for the CBR policy backup execution.
+* `backup_cycle` - (Required, List) Specifies the scheduling rule for the policy backup execution.
   The [object](#cbr_policy_backup_cycle) structure is documented below.
 
-* `enabled` - (Optional, Bool) Specifies whether to enable the CBR policy. Default to **true**.
+* `enabled` - (Optional, Bool) Specifies whether to enable the policy. Default to **true**.
```
Comment (houpeng80) [inline docs/resources/cbr_policy.md:69]:
```suggestion
* `enabled` - (Optional, Bool) Specifies whether to enable the policy. Defaults to **true**.
```

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_policy.go [hunk 3]
```diff
@@ -50,6 +58,7 @@ func ResourceCBRPolicyV3() *schema.Resource {
 				ValidateFunc: validation.StringInSlice([]string{
```
Comment (houpeng80) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_policy.go]: delete validation?

File: huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go [hunk 4]
```diff
@@ -23,13 +25,15 @@ func DataSourceCbrVaultsV3() *schema.Resource {
...
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The region in which to query the vaults.",
 			},
 			"name": {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ValidateFunc: validation.StringLenBetween(1, 64),
```
Comment (houpeng80) [inline huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go]: delete validation?

File: docs/data-sources/cbr_vaults.md (hunk 1 above)
Comment (Lance52259) [reply docs/data-sources/cbr_vaults.md:49]: emmmm...the forceNew description is incorrect

File: huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go]: okey

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_policy.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_policy.go]: okey.
BTW, for all regions, there are only backup and replication type of policies they have, it is different with the volume type

**Comment by Lance52259** [comment]
```
./scripts/codecheck.sh ./huaweicloud/services/cbr

==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        4      2153      147        32     1974        267            48.80
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~ices/cbr/resource_huaweicloud_cbr_vault.go       891       74        22      795        166            20.88
~ces/cbr/resource_huaweicloud_cbr_policy.go       415       37         6      372         65            17.47
~/cbr/data_source_huaweicloud_cbr_vaults.go       304       17         4      283         22             7.77
~/cbr/data_source_huaweicloud_cbr_backup.go       543       19         0      524         14             2.67
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     4      2153      147        32     1974        267            48.80
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 67330 bytes, 0.067 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
14 cbr resourceVaultCreate huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:411:1
13 cbr resourceVaultUpdate huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:802:1
9 cbr updateAssociatedResources huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:684:1
9 cbr waitForAllResourcesAssociated huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:650:1
9 cbr buildAssociateResourcesForServer huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:218:1
8 cbr resourcePolicyUpdate huaweicloud/services/cbr/resource_huaweicloud_cbr_policy.go:363:1
8 cbr resourcePolicyRead huaweicloud/services/cbr/resource_huaweicloud_cbr_policy.go:313:1
7 cbr updateBasicParameters huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:766:1
7 cbr flattenPolicies huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:536:1
7 cbr buildVaultCreateOpts huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:370:1
Average: 4.3

==> Checking for golangci-lint...

==> Checking for Nolint directives...

==> Checking for TF features in cbr...

==> Checking for misspell in cbr...

==> Checking for code complexity in ./huaweicloud/services/acceptance/cbr...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        4      1452       95         1     1356          4             1.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~br/resource_huaweicloud_cbr_policy_test.go       280       19         0      261          2             0.77
~cbr/resource_huaweicloud_cbr_vault_test.go       918       55         1      862          2             0.23
~data_source_huaweicloud_cbr_backup_test.go        74        7         0       67          0             0.00
~data_source_huaweicloud_cbr_vaults_test.go       180       14         0      166          0             0.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     4      1452       95         1     1356          4             1.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 45862 bytes, 0.046 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
2 cbr getVaultResourceFunc huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go:18:1
2 cbr getPolicyResourceFunc huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_policy_test.go:16:1
1 cbr testAccVault_bindPolicies_step2 huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go:891:1
1 cbr testAccVault_bindPolicies_step1 huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go:862:1
1 cbr testAccVault_bindPolicies_base huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go:802:1
Average: 1.05

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/cbr...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/cbr...

==> Cleanup patch...

Check Completed!
```

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAcc'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAcc -timeout 360m -parallel 4
=== RUN   TestAccDataBackup_basic
=== PAUSE TestAccDataBackup_basic
=== RUN   TestAccDataVaults_backupServer
=== PAUSE TestAccDataVaults_backupServer
=== RUN   TestAccDataVaults_replicationServer
=== PAUSE TestAccDataVaults_replicationServer
=== RUN   TestAccDataVaults_volume
=== PAUSE TestAccDataVaults_volume
=== RUN   TestAccDataVaults_backupTurbo
=== PAUSE TestAccDataVaults_backupTurbo
=== RUN   TestAccDataVaults_replicationTurbo
=== PAUSE TestAccDataVaults_replicationTurbo
=== RUN   TestAccPolicy_basic
=== PAUSE TestAccPolicy_basic
=== RUN   TestAccPolicy_retention
=== PAUSE TestAccPolicy_retention
=== RUN   TestAccPolicy_replication
=== PAUSE TestAccPolicy_replication
=== RUN   TestAccVault_backupServer
=== PAUSE TestAccVault_backupServer
=== RUN   TestAccVault_replicationServer
=== PAUSE TestAccVault_replicationServer
=== RUN   TestAccVault_prePaidServer
=== PAUSE TestAccVault_prePaidServer
=== RUN   TestAccVault_volume
=== PAUSE TestAccVault_volume
=== RUN   TestAccVault_backupTurbo
=== PAUSE TestAccVault_backupTurbo
=== RUN   TestAccVault_replicationTurbo
=== PAUSE TestAccVault_replicationTurbo
=== RUN   TestAccVault_AutoBind
=== PAUSE TestAccVault_AutoBind
=== RUN   TestAccVault_bindPolicies
=== PAUSE TestAccVault_bindPolicies
=== CONT  TestAccDataBackup_basic
=== CONT  TestAccVault_backupServer
=== CONT  TestAccVault_backupTurbo
=== CONT  TestAccDataVaults_replicationTurbo
--- PASS: TestAccDataVaults_replicationTurbo (17.91s)
=== CONT  TestAccVault_volume
--- PASS: TestAccVault_backupServer (384.82s)
=== CONT  TestAccVault_prePaidServer
--- PASS: TestAccVault_volume (381.70s)
=== CONT  TestAccVault_replicationServer
--- PASS: TestAccVault_replicationServer (13.58s)
=== CONT  TestAccVault_AutoBind
--- PASS: TestAccVault_AutoBind (18.40s)
=== CONT  TestAccVault_bindPolicies
--- PASS: TestAccVault_bindPolicies (33.00s)
=== CONT  TestAccDataVaults_volume
--- PASS: TestAccDataBackup_basic (658.84s)
=== CONT  TestAccDataVaults_backupTurbo
--- PASS: TestAccDataVaults_volume (340.02s)
=== CONT  TestAccDataVaults_replicationServer
--- PASS: TestAccDataVaults_replicationServer (14.01s)
=== CONT  TestAccVault_replicationTurbo
--- PASS: TestAccVault_replicationTurbo (12.64s)
=== CONT  TestAccDataVaults_backupServer
--- PASS: TestAccVault_prePaidServer (503.15s)
=== CONT  TestAccPolicy_retention
--- PASS: TestAccPolicy_retention (12.45s)
=== CONT  TestAccPolicy_replication
--- PASS: TestAccPolicy_replication (7.31s)
=== CONT  TestAccPolicy_basic
--- PASS: TestAccPolicy_basic (13.80s)
--- PASS: TestAccVault_backupTurbo (1010.81s)
--- PASS: TestAccDataVaults_backupServer (373.00s)
--- PASS: TestAccDataVaults_backupTurbo (581.45s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       1240.429s
```

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go [hunk 5]
```diff
@@ -228,62 +216,72 @@ func ResourceVault() *schema.Resource {
 }
 
 func buildAssociateResourcesForServer(rType string, resources []interface{}) ([]vaults.ResourceCreate, error) {
+	results := make([]vaults.ResourceCreate, len(resources))
 	if len(resources) == 0 {
-		return []vaults.ResourceCreate{}, nil
+		// The argument `resources` is required.
+		return results, nil
```
Comment (hmbig2) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: always return results， you can delete this line

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go [hunk 6]
```diff
@@ -771,13 +867,12 @@ func resourceVaultDelete(ctx context.Context, d *schema.ResourceData, meta inter
 		Refresh:    vaultStateRefreshFunc(client, d.Id()),
 		Timeout:    d.Timeout(schema.TimeoutDelete),
 		Delay:      5 * time.Second,
-		MinTimeout: 3 * time.Second,
+		MinTimeout: 5 * time.Second,
```
Comment (hmbig2) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: `PollInterval: 5 * time.Second,`

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: OFC

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go (hunk 6 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: done.
`5` -> `20`

**Pull request merged**